### PR TITLE
Fcore/inject reporter

### PR DESCRIPTION
**Pull request by arnauddieumegard**

### CODE REVIEW

File: Docgen/business/plugins/org.polarsys.kitalpha.doc.gen.business.core/templates/pattern._yyU7IvYiEd-jis7N5RhttA/method._yyU7J_YiEd-jis7N5RhttA.pt [hunk 1]
```diff
@@ -50,7 +50,7 @@
 	<div>
 	<p class="diagram-name" id="<%=id%>"><%= EscapeChars.forHTML(diagramName)%></p>
 	<p class="diagram">
-		<img id="<%=diagram.hashCode()%>" src="<%=generatedFolder%>/<%=imageFile.getName()%>" alt="<%= EscapeChars.forHTML(diagramName)%>" usemap="#<%=mapName%>" />
```
Comment (bzendagui) [inline Docgen/business/plugins/org.polarsys.kitalpha.doc.gen.business.core/templates/pattern._yyU7IvYiEd-jis7N5RhttA/method._yyU7J_YiEd-jis7N5RhttA.pt:53]: Are you sur to remove this data?  Where is it used ?

Comment (bzendagui) [reply Docgen/business/plugins/org.polarsys.kitalpha.doc.gen.business.core/templates/pattern._yyU7IvYiEd-jis7N5RhttA/method._yyU7J_YiEd-jis7N5RhttA.pt:53]: Please do not merge before answering my question ! Thanks.

Comment (arnauddieumegard) [reply Docgen/business/plugins/org.polarsys.kitalpha.doc.gen.business.core/templates/pattern._yyU7IvYiEd-jis7N5RhttA/method._yyU7J_YiEd-jis7N5RhttA.pt:53]: The id value (in line 51 and which value is the uid of the diagram - see value returned by DiagramSessionHelper.getID(EObject)) is the one used to reference the diagrams in the the diagram tree (see TreeService.printDiagramsTree for a DRepresentation element).
In the "Interested diagrams section" the reference is also done by referencing the "diagram-name" id (see CapellaServices.getImageLinkForDiagram).
The "Diagrams displaying" section relies on the getHyperLinkFromDiagram call which also relied only on the diagram uid.
Do you any other place where a diagram is referenced ?

Comment (arnauddieumegard) [reply Docgen/business/plugins/org.polarsys.kitalpha.doc.gen.business.core/templates/pattern._yyU7IvYiEd-jis7N5RhttA/method._yyU7J_YiEd-jis7N5RhttA.pt:53]: So to answer your question this data is never used.

Comment (arnauddieumegard) [reply Docgen/business/plugins/org.polarsys.kitalpha.doc.gen.business.core/templates/pattern._yyU7IvYiEd-jis7N5RhttA/method._yyU7J_YiEd-jis7N5RhttA.pt:53]: Your remark made me think about the "Interested diagrams" section. A new PR (https://github.com/eclipse/capella-xhtml-docgen/pull/166) has been created to fix a problem found.

Comment (bzendagui) [reply Docgen/business/plugins/org.polarsys.kitalpha.doc.gen.business.core/templates/pattern._yyU7IvYiEd-jis7N5RhttA/method._yyU7J_YiEd-jis7N5RhttA.pt:53]: What about the section "displayed in diagrams"? is it impacted by this modification?

Comment (arnauddieumegard) [reply Docgen/business/plugins/org.polarsys.kitalpha.doc.gen.business.core/templates/pattern._yyU7IvYiEd-jis7N5RhttA/method._yyU7J_YiEd-jis7N5RhttA.pt:53]: "displayed in diagrams" is the "Diagrams displaying" section.

Comment (bzendagui) [reply Docgen/business/plugins/org.polarsys.kitalpha.doc.gen.business.core/templates/pattern._yyU7IvYiEd-jis7N5RhttA/method._yyU7J_YiEd-jis7N5RhttA.pt:53]: Yes :)

**Review by tguiu — APPROVED** [review]

**Pull request merged**